R/get_biomart_pfam_merge.R

```diff
@@ -0,0 +1,120 @@
+#' Given pfam id(s) and an ensembl mart
+#' creates a grange for the pfam domains
+#' 
+#' @param ensembl a S4 class from biomaRt::useMart() call
+#' @param pfamDesc_path path to pfam description file
+#' @param ucscGenePfam_path path to ucsc gene pfam genomic location
+#' @param pfam_id pfam id(s) to filter
+#' @param return_pfam_gr logical value to return a pfam location granges instead of  dataframe
+#' 
+#' @export
+#'
+#' @return granges for genomic locations of given pfam id(s)
+#'
+#' @examples
+#' \dontrun{
+#' # hsapiens_gene_ensembl == Human genes (GRCh38.p13) 10th Feb 2021
+#' ensembl <- biomaRt::useMart(biomart = "ENSEMBL_MART_ENSEMBL", dataset = "hsapiens_gene_ensembl", host = "ensembl.org")
+#'
+#' download.file("http://hgdownload.soe.ucsc.edu/goldenPath/hg38/database/pfamDesc.txt.gz", "pfamDesc.txt.gz")
+#' download.file("http://hgdownload.soe.ucsc.edu/goldenPath/hg38/database/ucscGenePfam.txt.gz", "ucscGenePfam.txt.gz")
+#'
+#' bioMartDataPfam <- get_biomart_pfam_merge(ensembl = ensembl, pfamDesc_path = "pfamDesc.txt.gz", ucscGenePfam_path = "ucscGenePfam.txt.gz")
+#' }
+#'
+get_biomart_pfam_merge <- function(ensembl, pfamDesc_path, ucscGenePfam_path, pfam_id, return_pfam_gr = NULL) {
+  if (!missing(pfam_id)) {
+    dataBioMart <- biomaRt::getBM(
+      attributes = c(
+        "hgnc_symbol", "pfam",
+        "chromosome_name", "start_position", "end_position",
+        "strand"
+      ),
+      filters = "pfam",
+      values = pfam_id,
+      mart = ensembl
+    ) %>%
```

Comment:
We could have a gentle fail mechanism if (like not too rarely) biomart is not so responsive?